node/node.go

```diff
@@ -150,7 +150,7 @@ type Node struct {
 	transport   *p2p.MultiplexTransport
 	sw          *p2p.Switch  // p2p connections
 	addrBook    pex.AddrBook // known peers
-	nodeInfo    p2p.NodeInfo
+	nodeInfo    p2p.DefaultNodeInfo
```

Comment:
Given this is private to node it doesn't seem like it hurts. Also the RPC depends on `NodeInfo` being `DefaultNodeInfo` so seems to make sense to enforce the type.